### ISSUE TITLE
Bundler destroys rails installations on Windows

### ISSUE THREAD

**Issue by pkosenko** [bundler/bundler#5411]
After upgrading rails on Windows (Win 7), bundler keeps looking for a WINDOWS version of railties files that the rails project tells me DO NOT EXIST.  In fact, gem and bundler install the non-windows version of railties, then bundler complains that "windows" versions do not exist.  Then it refuses to allow the project to run.  This is a complete Ruby on Rails deal breaker for Windows, since it destroys one's ability to run or develop a project.  Bundler prevents Ruby on Rails from operating.

I thought that the problem might be with the 1.11.2 version of Bundler I was using, so I updated it to the latest 1.14.3

$ bundle -v
Bundler version 1.14.3
>gem list
railties (5.0.1, 4.2.5.1, 3.2.13, 3.2.0)  <-- NOTE . . . no windows platform versioning, but correct railties versions are installed.

$ rails server
Bundler could not find compatible versions for gem "railties":
  In Gemfile:
    rails (= 5.0.1) x86-mingw32 was resolved to 5.0.1, which depends on
      railties (= 5.0.1) x86-mingw32  <-- bundler is looking for a windows version
    sass-rails (~> 3.2.3) x86-mingw32 was resolved to 3.2.3, which depends on
      railties (~> 3.2.0.beta) x86-mingw32  <-- bundler is looking for a windows version

**Comment by segiddins** [comment]
Can you please share the information requested in https://github.com/bundler/bundler/blob/master/doc/contributing/ISSUES.md? Thanks!

**Comment by pkosenko** [comment]
Trial and error have helped me solve the problem.  I was MISLED by the ERROR MESSAGES.

The project I was working with had gems that depended on different versions of railties.  When I added those different versions of railties to the GEMFILE, I got a clear message that Bundler was unable to install different versions of the same GEM.  When I commented out the versions of SASS and CoffeeScript that depended on the OLDER railties (3.2.0), Bundler continued on with a "Bundle install" and ran "rails server".

So the issue is that the wording "Bundler could not find compatible versions for gem "railties"" is UNCLEAR.  It should have read "Bunder found INCOMPATIBLE versions of railties" or "Bundler cannot install different versions of railties (the same GEM)."  

And adding the Windows PLATFORM designation to the gem names led me on a wild goose chase of thinking that Bundler was looking for a Windows version of the railties gems (which do not exist).
